### PR TITLE
Serial queue

### PR DESCRIPTION
**Pull request by wom-bat**
Rename libsharedringbuffer.h to queue.h

### CODE REVIEW

**Comment by Ivan-Velickovic** [comment]
This will fail unless you update the sDDF submodule as well.

**Comment by wom-bat** [comment]
> This will fail unless you update the sDDF submodule as well.

I'm building to test inside LionsOS

File: examples/virtio/Makefile [hunk 1]
```diff
@@ -155,44 +161,35 @@ $(BUILD_DIR)/package_guest_images.o: $(VMM_TOOLS)/package_guest_images.S $(IMAGE
...
 
-$(BUILD_DIR)/%.o: %.c Makefile
-	$(CC) -c $(CFLAGS) $< -o $@
-
-$(BUILD_DIR)/%.o: $(VMM_SRC_DIR)/%.c Makefile
-	$(CC) -c $(CFLAGS) $< -o $@
-
-$(BUILD_DIR)/%.o: $(VMM_SRC_DIR)/util/%.c Makefile
-	$(CC) -c $(CFLAGS) $< -o $@
+VPATH=${VMM_SRC_DIR}:${VMM_SRC_DIR}/util:${VMM_SRC_DIR}/arch/aarch64:${SDDF_SERIAL_DRIVER}:${SDDF_UTIL}:${VMM_SRC_DIR}/arch/aarch64/vgic:${VMM_SRC_DIR}/virtio
```
Comment (Ivan-Velickovic) [inline examples/virtio/Makefile:164]: What is this for? I could not find it used anywhere, is it a special Make variable or something?

Comment (wom-bat) [reply examples/virtio/Makefile:164]: It's a standard Make thing to tell make where to look for source files.  Any precursor not in the current directory will be looked for in VPATH.  It allows removal of all the specific rules.

Comment (wom-bat) [reply examples/virtio/Makefile:164]: What's more, unlike addprefix, it's in POSIX make....

**Pull request merged**